bread/layout/components/datatable.py

```diff
@@ -133,20 +135,30 @@ def __init__(
                 hg.TH(headcontent, **getattr(col.header, "td_attributes", {}))
             )
 
-        self.iterator = hg.Iterator(
-            row_iterator,
-            rowvariable,
-            hg.TR(
-                *[
-                    hg.TD(col.cell, **getattr(col.cell, "td_attributes", {}))
-                    for col in columns
-                ]
-            ),
+        row_template = hg.TR(
+            *[
+                hg.TD(col.cell, **getattr(col.cell, "td_attributes", {}))
+                for col in columns
+            ]
         )
+
+        # combining Formsets with Datatables needs to handle iterations differently,
+        # the formsetfield needs to control the container in order to attach new entries/rows
+        if is_inlineformset:
+            # need local import here to prevent circular import
+            from .form import FormsetField
```

Comment:
Okay, continuing from above (because I suggested the opposite from your argument here :D).

If you use "isolated" and "dependent" instead of low-level and high-level the situation looks like this:

- form.py is highly dependant on Django implementation details (of the HTML forms). It is much more a logical component than a visual one.

- The other components should are very isolated and should only care about visual representation, not about django internals. Now, admittedly the datatable module has an ugly dependency to the models-system of Django for the from_model etc., but I would prefer to factor that out than to include even more logical dependencies in datatable.

Reply:
Ok, I get it :)
I am fine with form depending on DataTable and not the other way around.
But then, datatable should have a constructor or factory method which accepts a FormsetField. It should not create the FormsetField itself...